rewrite-core/src/main/java/org/openrewrite/config/YamlResourceLoader.java

```diff
@@ -146,8 +146,12 @@ public YamlResourceLoader(InputStream yamlInput,
                 .enable(MapperFeature.ACCEPT_CASE_INSENSITIVE_ENUMS)
                 .constructorDetector(ConstructorDetector.USE_PROPERTIES_BASED)
                 .build()
-                .registerModule(new ParameterNamesModule())
-                .disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
+                .registerModule(new ParameterNamesModule());
+
+        if (Boolean.FALSE.equals(properties.getOrDefault(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false))) {
```

Comment:
I'm not quite sure about the reuse of `Properties properties` for configuration; we at present use that to optionally replace properties in Yaml recipe files with looked up values.
https://github.com/openrewrite/rewrite/blob/83501012e8abaab57b9c006205dea7014db943c3/rewrite-core/src/main/java/org/openrewrite/config/YamlResourceLoader.java#L171-L172

I feel the dual use here might cause confusion in the future, but I'm also not yet sure about a better alternative. 

The prevailing style in `YamlResourceLoader` seems to be new overloaded constructors. Perhaps another one that takes a `Consumer<JsonMapper>` could offer the same functionality in a different way?

Reply:
Or we juse remove the line that says `disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES)` and see what catches fire. 🤷🏻

Reply:
One of the tests that fails when I try that quickly was added here
- https://github.com/openrewrite/rewrite/pull/3517

Reply:
Shannon rightfully pointed out that we might want to only and always enable `FAIL_ON_UNKNOWN_PROPERTIES` in tests, but keep that disabled when loading and running recipes in practice. Otherwise we get some unfortunate error handling we have to do in odd places. Just checking this in tests strikes the right balance we think, and ought to incentivize tests even more, similar to the type checks we do there.